sdk/storage/Azure.Storage.Common/src/Shared/StorageSharedKeyPipelinePolicy.cs

```diff
@@ -42,45 +42,46 @@ public override void OnSendingRequest(HttpMessage message)
         {
             base.OnSendingRequest(message);
 
+            var request = message.Request;
+
             // Add a x-ms-date header
             if (IncludeXMsDate)
             {
                 var date = DateTimeOffset.UtcNow.ToString("r", CultureInfo.InvariantCulture);
-                message.Request.Headers.SetValue(Constants.HeaderNames.Date, date);
+                request.Headers.SetValue(Constants.HeaderNames.Date, date);
             }
 
-            var stringToSign = BuildStringToSign(message);
+            var stringToSign = BuildStringToSign(request.Method, request.Uri.ToUri(), request.Content, request.Headers.ToDictionary(h => h.Name, h => h.Value));
             var signature = StorageSharedKeyCredentialInternals.ComputeSasSignature(_credentials, stringToSign);
 
             var key = new AuthenticationHeaderValue(Constants.HeaderNames.SharedKey, _credentials.AccountName + ":" + signature).ToString();
-            message.Request.Headers.SetValue(Constants.HeaderNames.Authorization, key);
+            request.Headers.SetValue(Constants.HeaderNames.Authorization, key);
         }
 
         // If you change this method, make sure live tests are passing before merging PR.
-        private string BuildStringToSign(HttpMessage message)
+        private string BuildStringToSign(RequestMethod requestMethod, Uri uri, RequestContent content, Dictionary<string, string> headers)
```

Comment:
Curious: how does this improve perf?

Reply:
It would be great to see the diffs in Benchmark.NET

Reply:
Removed because https://github.com/Azure/azure-sdk-for-net/pull/34195 solved the problem